lib/open-uri-cached-archive.rb

```diff
@@ -1,5 +1,43 @@
 require 'open-uri-cached-archive/version'
+require 'yaml'
+require 'stringio'
+require 'open-uri'
 
 class OpenUriCachedArchive
-  # Your code goes here...
+  class CachedResponse
+    def initialize(meta_file_path)
+      @meta = YAML.load_file(meta_file_path)
+      @body = File.read(meta_file_path.gsub(/\.meta$/, ''))
+      @status = meta.delete(:status)
+      @base_uri = meta.delete(:base_uri)
+      meta.delete(:content_type)
+    end
```

Comment:
Yep, good shout, I've pushed some commits to refactor things into methods. I've also changed the way I'm dealing with `meta`, and added a comment to the bit that was especially odd! Hopefully that's clearer now.